storage_controller/src/tenant_shard.rs

```diff
@@ -900,10 +911,63 @@ impl TenantShard {
                 // We only reach this point if there is work to do and we're going to skip
                 // doing it: warn it obvious why this tenant isn't doing what it ought to.
                 tracing::warn!("Skipping reconcile for policy {:?}", self.scheduling_policy);
-                return None;
+                return ReconcileNeeded::No;
             }
         }
 
+        ReconcileNeeded::Yes
+    }
+
+    /// Ensure the sequence number is set to a value where waiting for this value will make us wait
+    /// for the next reconcile: i.e. it is ahead of all completed or running reconcilers.
+    ///
+    /// Constructing a ReconcilerWaiter with the resulting sequence number gives the property
+    /// that the waiter will not complete until some future Reconciler is constructed and run.
+    fn advance_sequence(&mut self) {
```

Comment:
nit: I don't really like this name. I'd prefer `maybe_advance_sequence` to signal that it might no do anything.

Reply:
Yeah.  I've gone with `ensure_sequence_ahead`